### PR TITLE
Improve 2 of 20 most-visited GL JS Examples

### PR DESCRIPTION
**Pull request by domlet**
Re: https://github.com/mapbox/documentation/issues/654#

* Adjusts example content to improve visuals
* Elaborates on descriptions in plain language
* Adds links to API/Style Spec reference docs when mentioned 
* Adds code comments to increase accessibility
* Adds trailing slashes to links in this repo

## Before/after screenshots:

* `docs/pages/example/add-image.md`
* `docs/pages/example/geojson-polygon.md`

Screen Shot 2021-04-02 at 9 48 31 AM

### CODE REVIEW

**Comment by domlet** [comment]
## Troubleshooting link behavior

Here are three errors I corrected as suggested:

```
docs/pages/example/3d-buildings.md
20:78-20:144  error  Replace "/mapbox-gl-js/style-spec/layers#fill-extrusion" with "/mapbox-gl-js/style-spec/layers/#fill-extrusion" which adds a trailing slash to keep internal linking consistent  link-checker-skip-external  remark-lint-mapbox

docs/pages/example/animate-images.md
20:17-20:72  error  Replace "/mapbox-gl-js/style-spec/sources#image" with "/mapbox-gl-js/style-spec/sources/#image" which adds a trailing slash to keep internal linking consistent                  link-checker-skip-external  remark-lint-mapbox
  
docs/pages/example/heatmap-layer.md
19:52-19:108  error  Replace "/mapbox-gl-js/style-spec/layers#heatmap" with "/mapbox-gl-js/style-spec/layers/#heatmap" which adds a trailing slash to keep internal linking consistent                link-checker-skip-external  remark-lint-mapbox
```

When I run the site locally the link destinations don't work perfectly, as shown in this GIF:

funny-links

@katydecorah is there a solution for this local misbehavior, or something we can change in `link-checker-skip-external`  or `remark-lint-mapbox` to enable better linking?

**Comment by katydecorah** [comment]
> is there a solution for this local misbehavior, or something we can change in link-checker-skip-external or remark-lint-mapbox to enable better linking?

@domlet I think the problem is that the page shifts once the images on the page load. We saw a similar issue with https://docs.mapbox.com/api/maps/static-images/ after we added images. I can create a PR to add `height` attribtures to the images on this page to prevent the page from shifting.

File: docs/pages/example/add-image.html [hunk 1]
```diff
@@ -21,18 +28,20 @@
                                 'type': 'Feature',
                                 'geometry': {
                                     'type': 'Point',
-                                    'coordinates': [0, 0]
+                                    'coordinates': [-77.4144, 25.0759]
```
Comment (HeyStenson) [inline docs/pages/example/add-image.html:31]: Farewell, Null Island! 👋

**Review by HeyStenson — APPROVED** [review]
🎨 👍

**Comment by domlet** [comment]
Tagging @mapbox/gl-js team for review!

**Review by asheemmamoowala — APPROVED** [review]
LGTM! 🎉

**Pull request merged**